File: backend/src/main/java/com/flowlink/execution/dto/PendingWaitRequest.java
package com.flowlink.execution.dto;

/**
 * 콜백 대기(WAIT) 노드에서 실행이 중단됐을 때의 대기 정보.
 * 서버가 {@code url} 로 콜백을 받으면 스스로 재개하므로, 브라우저는 폴링으로 진행을 관전하며
 * {@code timeoutSec} 기준 카운트다운을 표시한다.
 */
public record PendingWaitRequest(
        String nodeId,
        String nodeName,
        String url,
        int timeoutSec
) {
}

File: backend/src/main/kotlin/com/flowlink/execution/dto/PendingWaitRequest.kt
package com.flowlink.execution.dto

/**
 * wait(콜백 대기) 노드에서 실행이 중단됐을 때 브라우저에 넘기는 대기 명세(표시·진행 상황용).
 * 콜백은 백엔드가 [receiveUrl] 로 직접 받아 재개하고([com.flowlink.execution.RelayController]),
 * [timeoutSec] 안에 콜백이 없으면 백엔드가 타임아웃으로 자동 재개(실행 FAILED)한다.
 *
 * [receiveUrl] 은 이 노드의 콜백 수신 URL({baseUrl}/relay/{execId}/cb/{nodeId}).
 */
data class PendingWaitRequest(
    val nodeId: String?,
    val nodeName: String?,
    val timeoutSec: Int,
    val receiveUrl: String?
)
